dough-data/src/test/java/io/github/bakedlibs/dough/collections/TestCopyUtils.java

```diff
@@ -0,0 +1,124 @@
+package io.github.bakedlibs.dough.collections;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+class TestCopyUtils {
+
+    private static <T> boolean haveSameElements(@Nonnull Collection<T> c1, @Nonnull Collection<T> c2) {
+        if (c1.size() != c2.size()) {
+            return false;
+        }
+        return c1.containsAll(c2);
+    }
+
+    private static <K, V> boolean haveSameElements(@Nonnull Map<K, V> m1, @Nonnull Map<K, V> m2) {
+        if (m1.size() != m2.size()) {
+            return false;
+        }
+        for (Map.Entry<K, V> entry : m1.entrySet()) {
+            if (!Objects.equals(m2.get(entry.getKey()), entry.getValue())) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private static <T> boolean haveClonedElements(@Nonnull Collection<T> c1, @Nonnull Collection<T> c2) {
+        if (c1.size() != c2.size()) {
+            return false;
+        }
+        for (T t1 : c1) {
+            for (T t2 : c2) {
+                // If they are the same, it did not clone and thus we fail.
+                if (t1 == t2) {
+                    return false;
+                } else if (t1.equals(t2)) {
+                    // If they are equal, the element has been cloned, thus, we break and check the next element.
+                    break;
+                }
+            }
+        }
+        return true;
+    }
+
+    private static <K, V> boolean haveClonedElements(@Nonnull Map<K, V> m1, @Nonnull Map<K, V> m2) {
+        if (m1.size() != m2.size()) {
+            return false;
+        }
+        for (Map.Entry<K, V> entry : m1.entrySet()) {
+            V otherValue = m2.get(entry.getKey());
+            // If they are the same, it did not clone and thus we fail.
+            if (entry.getValue() == otherValue) {
+                return false;
+            } else if (entry.getValue().equals(otherValue)) {
+                // If they are equal, the element has been cloned, thus, we break and check the next element.
+                break;
+            }
+        }
+        return true;
+    }
```

Comment:
All of these methods are used in `assertTrue` statements exclusively.
I think it would make sense to just convert these into assert methods themselves instead of returning a boolean.